packages/cosmic-swingset/src/block-manager.js

```diff
@@ -224,18 +224,19 @@ export default function makeBlockManager({
         verboseBlocks && console.info('block', action.blockHeight, 'begin');
 
         // Start a new block, or possibly replay the prior one.
-        for (const a of currentActions) {
-          // FIXME: This is a problem, apparently with Cosmos SDK.
-          // Need to diagnose.
-          if (a.blockHeight !== action.blockHeight) {
-            console.debug(
-              'Block',
-              action.blockHeight,
-              'begun with a leftover uncommitted action:',
-              a.type,
-            );
-          }
+        const leftover = currentActions
+          .filter(a => a.blockHeight !== action.blockHeight)
+          .map(a => a.type)
+          .join(', ');
+        if (leftover) {
+          // Leftover actions happen if queries or simulation are incorrectly
+          // resulting in accidental VM messages.
+          decohered = Error(
+            `Block ${action.blockHeight} begun with leftover uncommitted actions: ${leftover}`,
+          );
+          throw decohered;
```

Comment:
Does this crash _before_ the bad state gets committed or after? I think either design is acceptable, but I'd like to know which we're aiming at. Before is better, but it's still bad...

Reply:
Simulation happens between blocks.  So, it crashes before since no simulated transactions have yet been sent up to SwingSet before this point.